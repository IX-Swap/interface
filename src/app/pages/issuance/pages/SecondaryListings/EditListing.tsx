import React from 'react'
import { RootContainer } from 'ui/RootContainer'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { Grid } from '@mui/material'
import { ListingFormWrapper } from 'app/pages/issuance/components/ListingForm/ListingFormWrapper'
import { useParams } from 'react-router-dom'
import { useListingById } from 'app/pages/issuance/hooks/useListingById'
import { LoadingIndicator } from 'app/components/LoadingIndicator/LoadingIndicator'

export const EditListing = () => {
  const { listingId, issuerId } = useParams<{
    listingId: string
    issuerId: string
  }>()

  const { data, isLoading } = useListingById(listingId, issuerId)

  return (
    <Grid container direction='column' style={{ display: 'table' }}>
      <Grid item>
        <PageHeader
          title={'Edit Listing'}
          alignment='flex-start'
          showBreadcrumbs={true}
        />
      </Grid>
      <RootContainer>
        <Grid item>
          {isLoading ? (
            <LoadingIndicator />
          ) : (
            <ListingFormWrapper data={data} />
          )}
        </Grid>
      </RootContainer>
    </Grid>
  )
}
