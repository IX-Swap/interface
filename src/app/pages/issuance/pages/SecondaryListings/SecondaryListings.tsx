import React from 'react'
import { RootContainer } from 'ui/RootContainer'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { SecondaryListingsTable } from 'app/pages/issuance/components/SecondaryListingsTable/SecondaryListingsTable'
import { Grid } from '@mui/material'
import { TextInputSearchFilter } from 'app/components/TextInputSearchFilter'
import { AddListingButton } from 'app/pages/issuance/components/SecondaryListingsTable/AddListingButton'

export const SecondaryListings = () => {
  return (
    <Grid container style={{ display: 'table' }}>
      <PageHeader
        title={'Secondary Listings'}
        mainWrapperSX={{
          flexDirection: { xs: 'column', md: 'row' },
          alignItems: { xs: 'initial', md: 'normal' }
        }}
        endComponent={
          <Grid container spacing={2} wrap={'nowrap'}>
            <Grid item minWidth={320} width={'100%'}>
              <TextInputSearchFilter
                fullWidth
                placeholder='Search'
                inputAdornmentPosition='start'
                sx={{ height: 50 }}
              />
            </Grid>
            <Grid item width={'100%'}>
              <AddListingButton />
            </Grid>
          </Grid>
        }
      />
      <RootContainer>
        <SecondaryListingsTable />
      </RootContainer>
    </Grid>
  )
}
