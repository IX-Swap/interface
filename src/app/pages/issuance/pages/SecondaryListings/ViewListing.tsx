import { ListingDetails } from 'app/pages/issuance/components/ListingDetails/ListingDetails'
import { useListing, useListingOTC } from 'app/pages/invest/hooks/useListing'
import { Grid } from '@mui/material'
import React from 'react'
import { RootContainer } from 'ui/RootContainer'
import { PageHeader } from 'app/components/PageHeader/PageHeader'

export const ViewListing = () => {
  const { data, isLoading } = useListing()

  if (isLoading) {
    return null
  }

  return (
    <Grid container direction='column' spacing={4} display='table'>
      <Grid item>
        <PageHeader title={data?.tokenName} />
      </Grid>

      <Grid item style={{ paddingTop: 0 }}>
        <RootContainer>
          <Grid container spacing={6} wrap='wrap-reverse'>
            <Grid item xs={12}>
              <Grid container direction='column'>
                <Grid item>
                  <ListingDetails data={data} />
                </Grid>
              </Grid>
            </Grid>
          </Grid>
        </RootContainer>
      </Grid>
    </Grid>
  )
}

export const ViewListingOTC = () => {
  const { data, isLoading } = useListingOTC()

  if (isLoading) {
    return null
  }

  return <ListingDetails data={data} />
}
