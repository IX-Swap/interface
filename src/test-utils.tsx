import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
