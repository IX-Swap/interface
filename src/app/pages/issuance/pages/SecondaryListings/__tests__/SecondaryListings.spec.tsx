import React from 'react'
import { render } from 'test-utils'
import { SecondaryListings } from 'app/pages/issuance/pages/SecondaryListings/SecondaryListings'

describe('SecondaryListings', () => {
  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const container = render(<SecondaryListings />)
    expect(container).toMatchSnapshot()
  })
})
