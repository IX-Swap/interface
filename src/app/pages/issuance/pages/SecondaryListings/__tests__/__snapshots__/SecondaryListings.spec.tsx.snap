// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryListings should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        style="display: table;"
      >
        <div
          class="makeStyles-wrapper-5 MuiBox-root css-0"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-127nfur-MuiContainer-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-i9lcxu-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-4pkl32-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-mkpo7v"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1o4kjq1-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-geek62"
                      >
                        <h2
                          class="MuiTypography-root MuiTypography-h2 makeStyles-title-6 css-rldqdm-MuiTypography-root"
                        >
                          Secondary Listings
                        </h2>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <div
                        class="MuiBox-root css-ftbd7"
                      >
                        <nav
                          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-128kl2r-MuiTypography-root-MuiBreadcrumbs-root"
                        >
                          <ol
                            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                          />
                        </nav>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-uwwqev"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1cfjp91-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-1tii8sk-MuiGrid-root"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root makeStyles-root-9 css-1f0ts9k-MuiFormControl-root-MuiTextField-root"
                      >
                        <div
                          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-1ln009z-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-fuvw14-MuiSvgIcon-root"
                              data-testid="SearchIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                              />
                            </svg>
                          </div>
                          <input
                            aria-invalid="false"
                            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-mmb06o-MuiInputBase-input-MuiOutlinedInput-input"
                            id="mui-1"
                            placeholder="Search"
                            type="text"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-nnbavb"
                            >
                              <span
                                class="notranslate"
                              >
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-1mum6ye-MuiGrid-root"
                    >
                      <a
                        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-17s1bz2-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
                        href="/app/issuance/create"
                        tabindex="0"
                      >
                        Create a New Listing
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-p03apy-MuiContainer-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="makeStyles-loading-17 MuiBox-root css-0"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-q79gi7-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-11gelwl-MuiPaper-root"
                style="background-color: inherit;"
              >
                <div
                  class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
                  style="overflow: visible;"
                >
                  <table
                    aria-label="table"
                    class="MuiTable-root css-7mlt34-MuiTable-root"
                    data-testid="table"
                  >
                    <thead
                      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                      style="display: table-header-group;"
                    >
                      <tr
                        class="MuiTableRow-root MuiTableRow-head css-8dfetm-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                          scope="col"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                          >
                            <div
                              class="MuiBox-root css-d0uhtl"
                            >
                              Pair
                            </div>
                          </p>
                        </th>
                        <th
                          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                          scope="col"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                          >
                            Name
                          </p>
                        </th>
                        <th
                          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                          scope="col"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                          >
                            <div
                              class="MuiBox-root css-70qvj9"
                              style="cursor: pointer; user-select: none;"
                            >
                              Date
                              <div
                                class="MuiBox-root css-j7qwjs"
                                style="transform: rotate(180deg);"
                              />
                            </div>
                          </p>
                        </th>
                        <th
                          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                          scope="col"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                          >
                            Available Market
                          </p>
                        </th>
                        <th
                          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                          scope="col"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                          >
                            Status
                          </p>
                        </th>
                        <th
                          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                          scope="col"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                          >
                            Actions
                          </p>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                    />
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      style="display: table;"
    >
      <div
        class="makeStyles-wrapper-5 MuiBox-root css-0"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-127nfur-MuiContainer-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-i9lcxu-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-4pkl32-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-mkpo7v"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1o4kjq1-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-geek62"
                    >
                      <h2
                        class="MuiTypography-root MuiTypography-h2 makeStyles-title-6 css-rldqdm-MuiTypography-root"
                      >
                        Secondary Listings
                      </h2>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-ftbd7"
                    >
                      <nav
                        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-128kl2r-MuiTypography-root-MuiBreadcrumbs-root"
                      >
                        <ol
                          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                        />
                      </nav>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-uwwqev"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-1cfjp91-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-1tii8sk-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root makeStyles-root-9 css-1f0ts9k-MuiFormControl-root-MuiTextField-root"
                    >
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-1ln009z-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-fuvw14-MuiSvgIcon-root"
                            data-testid="SearchIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                            />
                          </svg>
                        </div>
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-mmb06o-MuiInputBase-input-MuiOutlinedInput-input"
                          id="mui-1"
                          placeholder="Search"
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-nnbavb"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-1mum6ye-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-17s1bz2-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
                      href="/app/issuance/create"
                      tabindex="0"
                    >
                      Create a New Listing
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-p03apy-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="makeStyles-loading-17 MuiBox-root css-0"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-q79gi7-MuiCircularProgress-root"
                role="progressbar"
                style="width: 40px; height: 40px;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-11gelwl-MuiPaper-root"
              style="background-color: inherit;"
            >
              <div
                class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
                style="overflow: visible;"
              >
                <table
                  aria-label="table"
                  class="MuiTable-root css-7mlt34-MuiTable-root"
                  data-testid="table"
                >
                  <thead
                    class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                    style="display: table-header-group;"
                  >
                    <tr
                      class="MuiTableRow-root MuiTableRow-head css-8dfetm-MuiTableRow-root"
                    >
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                        scope="col"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                        >
                          <div
                            class="MuiBox-root css-d0uhtl"
                          >
                            Pair
                          </div>
                        </p>
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                        scope="col"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                        >
                          Name
                        </p>
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                        scope="col"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                        >
                          <div
                            class="MuiBox-root css-70qvj9"
                            style="cursor: pointer; user-select: none;"
                          >
                            Date
                            <div
                              class="MuiBox-root css-j7qwjs"
                              style="transform: rotate(180deg);"
                            />
                          </div>
                        </p>
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                        scope="col"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                        >
                          Available Market
                        </p>
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                        scope="col"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                        >
                          Status
                        </p>
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headCell-13 css-1efwvfh-MuiTableCell-root"
                        scope="col"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 makeStyles-headText-14 css-1w73bwm-MuiTypography-root"
                        >
                          Actions
                        </p>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                  />
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
